tests/storage_pool/storage_pool_test.go

```diff
@@ -26,48 +26,104 @@ var _ = BeforeSuite(func() {
 	InitInstance()
 })
 
-var _ = Describe("{StoragePoolResize}", func() {
+var _ = Describe("{StoragePoolExpandDiskResize}", func() {
+
+	It("should get the existing pool and resize it by expanding a disk", func() {
 
-	It("should get the existing pool", func() {
+		var poolID string
 		pools, err := Inst().V.ListStoragePools(metav1.LabelSelector{})
 		if err != nil {
 			Expect(err).NotTo(HaveOccurred())
 		}
 
-		Expect(len(pools)).ShouldNot(BeEmpty())
+		Expect(len(pools)).NotTo(Equal(0))
 
-		var poolID string
 		for _, pool := range pools {
 			poolID = pool.Uuid
 			break
 		}
-		// Pick any pool and trigger pool resize command
 
 		poolToResize := pools[poolID]
+		Expect(poolToResize).ShouldNot(BeNil())
+		fmt.Println("POOL_TO_RESIZE: ", poolToResize)
 		oldSize := poolToResize.TotalSize
-		newSize := oldSize * 2
+		expectedSize := oldSize * 2 / units.GiB
 
-		err = Inst().V.ExpandPool(poolID, api.SdkStoragePool_RESIZE_TYPE_AUTO, newSize)
+		err = Inst().V.ExpandPool(poolID, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, expectedSize)
 		Expect(err).NotTo(HaveOccurred())
+		// wait for resize for 90 minutes, then fail
+		newTotalSize := uint64(0)
+
+		for i := 0; i < 60; i++ {
+			newPools, err := Inst().V.ListStoragePools(metav1.LabelSelector{})
+			Expect(err).NotTo(HaveOccurred())
+			var expandedPool *api.StoragePool
+			for _, pool := range newPools {
+				if poolID == pool.Uuid {
+					expandedPool = pool
+					break
+				}
+			}
 
-		time.Sleep(time.Second * 40)
+			Expect(expandedPool).ShouldNot(BeNil())
+			if expandedPool.TotalSize != poolToResize.TotalSize {
+				newTotalSize = expandedPool.TotalSize / units.GiB
+				break
+			}
+			time.Sleep(time.Second * 90)
+		}
+		Expect(newTotalSize).NotTo(BeNumerically("==", 0), "Pool size should not be 0. Pool expand operation has failed")
+		Expect(newTotalSize).Should(BeNumerically("==", expectedSize), fmt.Sprintf("expected new size '%d' to be equal to the expected size  '%d'", newTotalSize, expectedSize))
```

Comment:
move these steps to the volume driver like ValidateStoragePools and use that method in each test

Reply:
This looks more like a helper function and we should not mix it with the driver behavior. But I agree, I'll remove the code duplication